### DISCUSSION TITLE
Upload to - expected behaviour

### DISCUSSION THREAD

**Discussion by jakobrosenberg** [pycom/pymakr-vsc#222] [Q&A] [answered]
Currently the project view's `upload to device` button does just that. Uploads.

How should we handle deletion of files when uploading?

Option A
Add a `sync to device` which clones the project

Option B
Same as A, but also remove the old `upload to device`

Option C
Prompt on upload

---

Personally I prefer option B for the project view and then reserving `upload to device` for the file explorer context menu.

Eg. `sync to device` and `sync from device`:

![image](https://user-images.githubusercontent.com/4153004/162935271-86974a6b-cd27-40f0-9c9d-27caed679ed1.png)

and `upload to device`:

![image](https://user-images.githubusercontent.com/4153004/162935460-63be4ffd-c63f-4716-abd0-512c2dca72e8.png)

**Comment by Josverl** [comment, marked as answer]
On a project level I think a sync (option B) makes sense.
But there should also be an option to download a file or folder recursively.
In my micropython-stubber project I depend on the ability to download a folder of files that are generated on the MCU
 Explorer view seems a good place for that.
And in the interest of symmetry an upload for a file/ folder then also makes sense.

**Reply by jakobrosenberg** [reply]
I've added the `sync to device` in our current `next-staging` branch, but have left out `sync from device`. The latter doesn't seem to solve anything and source files should always be deleted manually. For now I think we should stick with the existing download-from-device behavior.
